### ISSUE TITLE
Create a sequence diagram that shows how a certificate gets issued with let's encrypt

### ISSUE THREAD

**Issue by maelvls** [cert-manager/website#426]
During the "cert-manager: 2020 retrospective" on 6th January 2021 (see [retro dashboard](https://action.parabol.co/team/hshPyuEuf)), we discussed the need for a diagram about the "internals" of cert-manager.

The goal is to represent a well-known flow (the creation of a Certificate using Let's Encrypt) and draw all the cert-manager CRDs and Secrets related to this flow.

I had drawn a quick diagram [a long time ago](https://kubernetes.slack.com/archives/CDEQJ0Q8M/p1604673834216600):

<img alt="" src="https://user-images.githubusercontent.com/2195781/113146120-96706a80-922f-11eb-8348-bc9102656dc0.jpg" width="300">

I then started converting the drawing to a [textik diagram](https://textik.com/#7dc27a1faa9d56de0), but I soon realized it wouldn't be as practical as I intended it to be:

<img alt="" src="https://user-images.githubusercontent.com/2195781/107038034-3d9acd80-67bc-11eb-84f4-a208cd67a6b7.png" width="300">

As @coderanger suggested a month ago, there might be a better way with https://mermaid-js.github.io/mermaid/#/sequenceDiagram.

Irbe's high-level diagram: https://miro.com/app/board/o9J_lUdPqQI=/

Irbe's great [diagram](https://kubernetes.slack.com/archives/CDEQJ0Q8M/p1617730470255500?thread_ts=1617703491.239900&cid=CDEQJ0Q8M):

<img alt="" src="https://user-images.githubusercontent.com/2195781/114370211-820d6580-9b7f-11eb-944e-37a2d3add419.jpg" width="300">

**Comment by irbekrm** [comment]
> During the "cert-manager: 2020 retrospective" on 6th January 2021 (see retro dashboard), we discussed the need for a diagram about the "internals" of cert-manager.
I started drawing a diagram using the ASCII editor textik.com, but I soon realized it wouldn't be as practical as I intended it to be

Maybe this could be a cool talk (for example for cert-manager dev days) with a few diagrams, walk through the codebase etc.

**Comment by maelvls** [comment]
Progress update: I abandonned the idea of using a sequence diagram, it just makes things even harder to understand. Here is what I have now:

![IMG_6515](https://user-images.githubusercontent.com/2195781/115728505-d89b5080-a384-11eb-94e7-1dc1f359a527.jpeg)

![IMG_6514](https://user-images.githubusercontent.com/2195781/115728488-d5a06000-a384-11eb-846f-f043f7fb4130.jpeg)

**Comment by irbekrm** [comment]
> Progress update: I abandoned the idea of using a sequence diagram, it makes things harder to understand. Here is what I have now:
I've not followed the sequence through, but really like the visual of this 😄

**Comment by maelvls** [comment]
Related: #568

**Comment by maelvls** [comment]
Today I made another attempt!

- Still too much information/bloated,
- Some mistakes e.g. the red arrow at the bottom-left, it should originate from the certificate request,
- I did not show the webhooks interactions.

I'll try to split that into two whiteboards to see.

![diagram](https://user-images.githubusercontent.com/2195781/119226008-68a8f300-bb07-11eb-840a-39cf4f1b2be4.jpg)

**Comment by maelvls** [comment]
Source: [letsencrypt-flow-cert-manager.drawio.zip](https://github.com/cert-manager/website/files/6546309/letsencrypt-flow-cert-manager.drawio.zip)

<img alt="letsencrypt-flow-cert-manager" src="https://user-images.githubusercontent.com/2195781/119652066-f1849f00-be25-11eb-8649-9287f1a6a455.png" width="800">

**Comment by maelvls** [comment]
Thoughts after our discussion with @irbekrm this afternoon:

1. Mentioning "issuing controller", "keymanager controller", "requestmanager", "acmeissuer" is quite confusing for a newcomer. It would be good for cert-manager developers, but we agreed that this diagram would be more helpful without these controller details. I'll say "cert-manager" everywhere.
2. The Ingress, Pod and Service created by cert-manager for the HTTP01 challenge is not shown here, and these would really help understand the whole flow. I'll try to add them.
3. The details about `revision: nil` and `cert-manager.io/certificate-revision: "1"` are only helpful for people who are trying to understand the link between certificates and certificate requests. I proposed to move the diagram in [gatherer.go](https://github.com/jetstack/cert-manager/blob/868d2f1637be6a3ae7fee1078ae996a8acd5aca2/pkg/controller/certificates/trigger/policies/gatherer.go) to a real diagram that we show on the website.

**Comment by maelvls** [comment]
I just realized that I have only shared a PNG file (using Google Drive) which embedded the source drawio XML. So I opened that PNG in Drawio and exported it to a .drawio file here:

[letsencrypt-flow-cert-manager.drawio.zip](https://github.com/cert-manager/website/files/12545952/letsencrypt-flow-cert-manager.drawio.zip)

Let's add that directly to the website repository along side the PNG file.

I'll close this issue.
